src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -577,6 +577,58 @@ public DirectAudioControllerImpl getDirectAudioController()
                 });
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Channel> retrieveChannelById(long channelId)
```

Comment:
Given that we are likely not going to expose this publicly yet, we can kill off the handling of anything but private channel and move the impl into PrivateChannelImpl.